Orange/widgets/visualize/utils/plotgui.py

```diff
@@ -0,0 +1,113 @@
+from AnyQt.QtWidgets import QAction
+from AnyQt.QtCore import Qt
+from AnyQt.QtGui import QKeySequence
+
+from Orange.widgets import gui
+
+
+JITTER_SIZES = [0, 0.1, 0.5, 1, 2, 3, 4, 5, 7, 10]
+
+
+def jittering(master, parent, jitter_sizes=None, is_cb_jitter_cont=False):
+    jitter_sizes = jitter_sizes if isinstance(jitter_sizes, list) else JITTER_SIZES
+    gui.valueSlider(
+        widget=parent, master=master, value='graph.jitter_size',
+        label="Jittering: ", values=jitter_sizes, callback=master.reset_graph_data,
+        labelFormat=lambda x:
+        "None" if x == 0 else ("%.1f %%" if x < 1 else "%d %%") % x)
+    if is_cb_jitter_cont:
+        gui.checkBox(
+            widget=gui.indentedBox(widget=parent),
+            master=master, value="graph.jitter_continuous",
+            label="Jitter numeric values", callback=master.reset_graph_data)
+
+def box_axis_data(master, parent):
+    common_options = dict(
+        labelWidth=50, orientation=Qt.Horizontal, sendSelectedValue=True,
+        valueType=str)
+    box_axis_data = gui.vBox(parent, "Axis Data")
+    master.cb_attr_x = gui.comboBox(
+        box_axis_data, master, "attr_x", label="Axis x:", callback=master.update_attr,
+        model=master.xy_model, **common_options)
+    master.cb_attr_y = gui.comboBox(
+        box_axis_data, master, "attr_y", label="Axis y:", callback=master.update_attr,
+        model=master.xy_model, **common_options)
+    return box_axis_data
+
+def box_radius(master, parent):
+    box_radius = gui.widgetBox(parent, "Hide")
+    master.rslider = gui.hSlider(
+        widget=box_radius, master=master, value="radius", minValue=0, maxValue=100,
+        step=5, label="Radius", createLabel=False, ticks=True,
+        callback=master.update_radius)
+    master.rslider.setTickInterval(0)
+    master.rslider.setPageStep(10)
+    return box_radius
+
+def box_plot_properties(master, parent):
+    box_plot_prop = gui.vBox(parent, "Plot Properties")
+    g = master.graph.gui
+    g.add_widgets([g.ShowLegend, g.ShowGridLines], box_plot_prop)
+    gui.checkBox(
+        widget=box_plot_prop, master=master, value="graph.tooltip_shows_all",
+        label="Show all data on mouse hover")
+    master.cb_class_density = gui.checkBox(
+        widget=box_plot_prop, master=master, value="graph.class_density",
+        label='Show class density',
+        callback=master.update_density)
+    master.cb_reg_line = gui.checkBox(
```

Comment:
Put these attributes on the box, e.g. `box_plot_prop.cb_reg_lin = gui.checkBox(...`.

Even better, don't store it at all: *explore whether* checkbox can be accessed through `self.controls.graph.class_density` and through `self.graph.controls.class_density`.